io.openems.edge.battery.pylontech/test/io/openems/edge/battery/pylontech/powercubem2/PylontechPowercubeM2BatteryImplTest.java

```diff
@@ -0,0 +1,101 @@
+package io.openems.edge.battery.pylontech.powercubem2;
+
+import io.openems.common.types.ChannelAddress;
+import io.openems.edge.common.startstop.StartStopConfig;
+import io.openems.edge.common.test.AbstractComponentTest.TestCase;
+import io.openems.edge.battery.pylontech.powercubem2.statemachine.StateMachine;
+import io.openems.edge.bridge.modbus.test.DummyModbusBridge;
+import io.openems.edge.common.test.ComponentTest;
+import io.openems.edge.common.test.DummyComponentManager;
+import io.openems.edge.common.test.DummyConfigurationAdmin;
+import io.openems.edge.common.test.TimeLeapClock;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import org.junit.Test;
+
+/**
+ * Checks that the battery wakes up correctly and progresses
+ * through the state machine in response to commands.
+ */
+public class PylontechPowercubeM2BatteryImplTest {
+
+	private static final String BATTERY_ID = "battery0";
+	private static final String MODBUS_ID = "modbus0";
+
+	private static final ChannelAddress STATE_MACHINE = new ChannelAddress(BATTERY_ID,
+			PylontechPowercubeM2Battery.ChannelId.STATE_MACHINE.id());
+	private static final ChannelAddress STATUS_CHANNEL = new ChannelAddress(BATTERY_ID,
+			PylontechPowercubeM2Battery.ChannelId.SYSTEM_STATUS.id());
+
+	@Test
+	public void wakeUpBattery() throws Exception {
+
+		final var clock = new TimeLeapClock(Instant.parse("2020-01-01T01:00:00.00Z"), ZoneOffset.UTC);
```

Comment:
Is the clock necessary? You can construct a `DummyComponentManager` without a clock too. The same for the other test and the `startBatteryTest` in the other file.

Reply:
Yes - this was unnecessary